### ISSUE TITLE
Oops: Installation failed

### ISSUE THREAD

**Issue by LusiXie** [plloydsmith/rmdcev#1]
I suppose it has something to do with 'Rcpp' and 'rlang' because the installation skips these two? 

Here is what I get starting with a clean R session: 

> install_github("plloydsmith/rmdcev")
Downloading GitHub repo plloydsmith/rmdcev@master
from URL https://api.github.com/repos/plloydsmith/rmdcev/zipball/master
Installing rmdcev
Installing 1 package: dplyr

  There is a binary version available (and will be installed) but the source version is later:
      binary  source
dplyr  0.7.8 0.8.0.1

also installing the dependencies ‘Rcpp’, ‘rlang’

trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.5/Rcpp_1.0.0.zip'
Content type 'application/zip' length 4480048 bytes (4.3 MB)
downloaded 4.3 MB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.5/rlang_0.3.1.zip'
Content type 'application/zip' length 1076832 bytes (1.0 MB)
downloaded 1.0 MB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.5/dplyr_0.7.8.zip'
Content type 'application/zip' length 3069506 bytes (2.9 MB)
downloaded 2.9 MB

package ‘Rcpp’ successfully unpacked and MD5 sums checked
package ‘rlang’ successfully unpacked and MD5 sums checked
package ‘dplyr’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\lusix\AppData\Local\Temp\Rtmp8OgPdy\downloaded_packages
Installing 1 package: purrr
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.5/purrr_0.3.0.zip'
Content type 'application/zip' length 413749 bytes (404 KB)
downloaded 404 KB

package ‘purrr’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\lusix\AppData\Local\Temp\Rtmp8OgPdy\downloaded_packages
Skipping install of 'Rcpp' from a cran remote, the SHA1 (1.0.0) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'rlang' from a cran remote, the SHA1 (0.3.1) has not changed since last install.
  Use `force = TRUE` to force installation
Installing 1 package: stringr
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.5/stringr_1.4.0.zip'
Content type 'application/zip' length 215310 bytes (210 KB)
downloaded 210 KB

package ‘stringr’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\lusix\AppData\Local\Temp\Rtmp8OgPdy\downloaded_packages
"D:/Programs/R-3.5.0/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  "C:/Users/lusix/AppData/Local/Temp/Rtmp8OgPdy/devtools4a2c367f47ed/plloydsmith-rmdcev-f7d3318" --library="D:/Programs/R-3.5.0/library"  \
  --install-tests 

* installing *source* package 'rmdcev' ...
** libs
Error in .shlib_internal(args) : 
  C++14 standard requested but CXX14 is not defined
* removing 'D:/Programs/R-3.5.0/library/rmdcev'
In R CMD INSTALL
Installation failed: Command failed (1)

**Comment by plloydsmith** [comment]
I think its the call to C++14. In your Makevars file of the .R folder (usually in your computer user folder) ensure the first two lines are

CXX14 = $(BINPREF)g++ -m$(WIN) -std=c++1y
CXX14FLAGS=-O3 -mtune=native -march=native

I think installing rstan first and then rmdcev might be recommended.

**Comment by LusiXie** [comment]
Issue solved - you can close it now. 
It wasn't required when I installed rstan. I made myself a note. 
Thanks.

**Comment by plloydsmith** [comment]
Great, first issue a success.

**Comment by LusiXie** [comment]
Sorry the installation issue again.

As my laptop is slow in simulations, I was trying to use the desktops at lab. I followed the exactly same procedure to two desktops as I did for the laptop, but several issues occurred. 

#1 Default (i.e. same procedure as for the laptop)
library(devtools)
install_github("plloydsmith/rmdcev")

* installing *source* package 'rmdcev' ...
** libs

*** arch - i386
make: Nothing to be done for `all'.
installing to C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/i386

*** arch - x64
make: Nothing to be done for `all'.
installing to C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/x64
** R
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
  converting help for package 'rmdcev'
    finding HTML links ... done
    CheckMdcevData                          html  
    CreateBlankPolicies                     html  
    CreateListsCol                          html  
    CreateListsRow                          html  
    CreatePsiMatrix                         html  
    DoCbind                                 html  
    FitMDCEV                                html  
    GrabParms                               html  
    HierarchicalBayesMDCEV                  html  
    MultiplyMatrix                          html  
    PrepareSimulationData                   html  
    ReduceStanFitSize                       html  
    RunStanSampling                         html  
    SimulateMdcevData                       html  
    SimulateWTP                             html  
    StanWTP                                 html  
    SummaryWelfare                          html  
    maxlikeMDCEV                            html  
    processMDCEVdata                        html  
    rmdcev                                  html  
** building package indices
** testing if installed package can be loaded
*** arch - i386
**Error: package or namespace load failed for 'rmdcev' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/i386/rmdcev.dll':**
  **LoadLibrary failure:  %1 is not a valid Win32 application.**

Error: loading failed
Execution halted
*** arch - x64
Error: package or namespace load failed for 'rmdcev' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/x64/rmdcev.dll':
  LoadLibrary failure:  The specified procedure could not be found.

Error: loading failed
Execution halted
ERROR: loading failed for 'i386', 'x64'
* removing 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev'
In R CMD INSTALL
Installation failed: Command failed (1)

#2 install_github("plloydsmith/rmdcev", lib="C:/Program Files/R/R-3.4.4/library", args='--no-multiarch')

Error: package or namespace load failed for 'rmdcev' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/x64/rmdcev.dll':
  LoadLibrary failure:  The specified procedure could not be found.

#3 install_github("plloydsmith/rmdcev", lib="C:/Program Files/R/R-3.4.4/library", args=c('--no-multiarch','--no-test-load'))
Ended with: (No error message) 
* DONE (rmdcev)
In R CMD INSTALL

When I was trying to load the package, I got this: 
Error: package or namespace load failed for ‘rmdcev’ in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/x64/rmdcev.dll':
  LoadLibrary failure:  The specified procedure could not be found.
(same error message as above) plus: 
The procedure entry point R_ContinueUnwind could not be located in the dynamic link library.......

The R Versions are similar, i.e. platform, arch and os, etc.
 
R.Version()
$platform
[1] "x86_64-w64-mingw32"

$arch
[1] "x86_64"

$os
[1] "mingw32"

$system
[1] "x86_64, mingw32"

...

Should I install the package as a administor instead of a user? 

Thanks.

Best,
Lusi

**Comment by plloydsmith** [comment]
Hi Lusi,

Weird, I think you were on the right track using the install_github options. I think it might be the R version 3.4 versus 3.5 in which case I should update the dependencies. I'll take a look at that this week.

One question, have you compared any model and welfare results across this version compared to the matlab or older R version. Are they giving your data the same results? They did for the data I was using but would like to test it on other data sets.

Pat

Patrick Lloyd-Smith
Assistant Professor
Department of Agricultural and Resource Economics
Global Institute for Water Security
College of Agriculture and Bioresources
University of Saskatchewan
Tel. (306) 966-8706




On Wed, 20 Feb 2019 at 11:53, Lusi Xie <notifications@github.com<mailto:notifications@github.com>> wrote:

Sorry the installation issue again.

As my laptop is slow in simulations, I was trying to use the desktops at lab. I followed the exactly same procedure to two desktops as I did for the laptop, but several issues occurred.

#1<https://github.com/plloydsmith/rmdcev/issues/1> Default (i.e. same procedure as for the laptop)
library(devtools)
install_github("plloydsmith/rmdcev")

  *   installing source package 'rmdcev' ...
** libs

*** arch - i386
make: Nothing to be done for `all'.
installing to C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/i386

*** arch - x64
make: Nothing to be done for `all'.
installing to C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/x64
** R
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
converting help for package 'rmdcev'
finding HTML links ... done
CheckMdcevData html
CreateBlankPolicies html
CreateListsCol html
CreateListsRow html
CreatePsiMatrix html
DoCbind html
FitMDCEV html
GrabParms html
HierarchicalBayesMDCEV html
MultiplyMatrix html
PrepareSimulationData html
ReduceStanFitSize html
RunStanSampling html
SimulateMdcevData html
SimulateWTP html
StanWTP html
SummaryWelfare html
maxlikeMDCEV html
processMDCEVdata html
rmdcev html
** building package indices
** testing if installed package can be loaded
*** arch - i386
Error: package or namespace load failed for 'rmdcev' in inDL(x, as.logical(local), as.logical(now), ...):
unable to load shared object 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/i386/rmdcev.dll':
LoadLibrary failure: %1 is not a valid Win32 application.

Error: loading failed
Execution halted
*** arch - x64
Error: package or namespace load failed for 'rmdcev' in inDL(x, as.logical(local), as.logical(now), ...):
unable to load shared object 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/x64/rmdcev.dll':
LoadLibrary failure: The specified procedure could not be found.

Error: loading failed
Execution halted
ERROR: loading failed for 'i386', 'x64'

  *   removing 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev'
In R CMD INSTALL
Installation failed: Command failed (1)

#2 install_github("plloydsmith/rmdcev", lib="C:/Program Files/R/R-3.4.4/library", args='--no-multiarch')

Error: package or namespace load failed for 'rmdcev' in inDL(x, as.logical(local), as.logical(now), ...):
unable to load shared object 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/x64/rmdcev.dll':
LoadLibrary failure: The specified procedure could not be found.

#3 install_github("plloydsmith/rmdcev", lib="C:/Program Files/R/R-3.4.4/library", args=c('--no-multiarch','--no-test-load'))
Ended with: (No error message)

  *   DONE (rmdcev)
In R CMD INSTALL

When I was trying to load the package, I got this:
Error: package or namespace load failed for ‘rmdcev’ in inDL(x, as.logical(local), as.logical(now), ...):
unable to load shared object 'C:/Users/lxie/Documents/R/win-library/3.4/rmdcev/libs/x64/rmdcev.dll':
LoadLibrary failure: The specified procedure could not be found.
(same error message as above) plus:
The procedure entry point R_ContinueUnwind could not be located in the dynamic link library.......

The R Versions are similar, i.e. platform, arch and os, etc.

R.Version()
$platform
[1] "x86_64-w64-mingw32"

$arch
[1] "x86_64"

$os
[1] "mingw32"

$system
[1] "x86_64, mingw32"

...

Should I install the package as a administor instead of a user?

Thanks.

Best,
Lusi

**Comment by LusiXie** [comment]
Hi Pat,

I forgot to mention that the two desktops have different versions of R (one is 3.4 and another one is 3.5). I wonder if it has something to do with the user type (needs to be the admin user?) as these are REES desktops. (I'm currently running it on my laptop and sometimes it gets crazy. I suppose desktops should be better.)

The estimates and welfare results I got from the package are very close to the old R version - not the same every time but it is the case with the old R version. Matlab had the convergence issue (i.e. 0 se's) so I didn't trust the results. 

If you still remember, I once encountered an issue with estimation of spatial KT using four other SP scenarios - not anymore with the package version. Not sure why though. But I got estimates and welfare measures. 

Best,
Lusi

**Comment by plloydsmith** [comment]
Hmm, yah it might be the admin type (although I'd like the package to work with either). Glad to hear the results are robust.

Pat

Patrick Lloyd-Smith
Assistant Professor
Department of Agricultural and Resource Economics
Global Institute for Water Security
College of Agriculture and Bioresources
University of Saskatchewan
Tel. (306) 966-8706




On Tue, 26 Feb 2019 at 19:12, Lusi Xie <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi Pat,

I forgot to mention that the two desktops have different versions of R (one is 3.4 and another one is 3.5). I wonder if it has something to do with the user type (needs to be the admin user?) as these are REES desktops. (I'm currently running it on my laptop and sometimes it gets crazy. I suppose desktops should be better.)

The estimates and welfare results I got from the package are very close to the old R version - not the same every time but it is the case with the old R version. Matlab had the convergence issue (i.e. 0 se's) so I didn't trust the results.

If you still remember, I once encountered an issue with estimation of spatial KT using four other SP scenarios - not anymore with the package version. Not sure why though. But I got estimates and welfare measures.

Best,
Lusi
